js/app/controllers/calcontroller.js

```diff
@@ -140,8 +140,10 @@ app.controller('CalController', ['$scope', 'Calendar', 'CalendarService', 'VEven
 				$scope.$apply();
 			});
 		} else {
-			$scope.calendarsPromise = CalendarService.getPublicCalendar(constants.publicSharingToken).then(function(calendar) {
-				$scope.calendars = [calendar];
+			constants.publicSharingToken.split(".").forEach( (token) => {
+				$scope.calendarsPromise = CalendarService.getPublicCalendar(token).then(function(calendar) {
+					$scope.calendars.push(calendar);
+			});
 				is.loading = false;
 				// TODO - scope.apply should not be necessary here
 				$scope.$apply();
```

Comment:
These two lines are still called for each calendar. Please only call them once all calendars loaded.

To make sure they are called after all are done, you need to create an array with all `CalendarService.getPublicCalendar(token)` promises and do a `Promise.all()` on the array.

For more information about Promise functions see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Reply:
Furthermore the `catch` is currently called on the `forEach` function. It has to be called on the Promise instead.

Reply:
I will look into the same :)